internal/provider/virtualmachine/resource_virtualmachine_constructor.go

```diff
@@ -122,6 +122,15 @@ func (c *Constructor) Setup() util.Processors {
 				return nil
 			},
 		},
+		{
+			Field: constants.FieldVirtualMachineReservedMemory,
+			Parser: func(i interface{}) error {
+				vmBuilder.Annotations(map[string]string{
+					harvesterutil.AnnotationReservedMemory: i.(string),
+				})
+				return nil
+			},
```

Comment:
```suggestion
			Field: constants.FieldVirtualMachineReservedMemory,
			Parser: func(i interface{}) error {
				reservedMemory := i.(string)
				if reservedMemory != "" {
					vmBuilder.Annotations(map[string]string{
						harvesterutil.AnnotationReservedMemory: i.(string),
					})
				} else {
					delete(vmBuilder.VirtualMachine.Annotations, harvesterutil.AnnotationReservedMemory)
				}
				return nil
			},
			Required: true,

Reply:
need consider and test this case: change the reseved memory from 100Mi to "" or remove the "reserved_memory" field

Reply:
Appreciated for the test and consideration. I have updated according to the suggestion